### PR TITLE
Added the Overview of modules assembly to the pantheon2.yml file.

### PR DESCRIPTION
**Pull request by adahms**
Adding the "Overview of Modules" assembly and modules to the pantheon2.yml file.

Also made several minor adjustments, such as deleting an old version of the 'Definition of a module' module from the 'modules' directory and ensuring several file names and anchor IDs were all consistent with the title of the module and the proposed table of contents.

After these are merged, we can link to the content from our main help assembly.

### CODE REVIEW

**Comment by adahms** [comment]
@redhataccess/pantheon-developers - No technical changes to the content itself, but requesting review of the changes to the pantheon2.yml file just in case.

**Comment by benradey** [comment]
Looks solid from my perspective!

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-content.adoc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //include::../attributes.adoc[]
-[id="publishing-content-in-pantheon_{context}"]
+[id="publishing-content_{context}"]
 = Publishing {ContentTerm}
 
 [role=_abstract]
```
Comment (theashiot) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-content.adoc:5]: should be [role="_abstract"]

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_unpublishing-content.adoc [hunk 2]
```diff
@@ -1,5 +1,5 @@
 //include::../attributes.adoc[]
-[id="unpublishing-content-in-pantheon_{context}"]
+[id="unpublishing-content_{context}"]
 = Unpublishing {ContentTerm}
 
 [role=_abstract]
```
Comment (theashiot) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_unpublishing-content.adoc:5]: should be [role="_abstract"]

**Comment by theashiot** [comment]
Hi @adahms,

Your changes look good. I noticed that in a couple of places double quotes are missing in role tag - not in tour changes tohugh. I have commented inline. 

best,
ashwin

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_unpublishing-content.adoc (hunk 2 above)
Comment (adahms) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_unpublishing-content.adoc]: Updated.

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-content.adoc (hunk 1 above)
Comment (adahms) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/proc_publishing-content.adoc]: Updated.

**Comment by adahms** [comment]
Thanks for the review, @theashiot - it's ready for another look.

**Review by theashiot — APPROVED** [review]

**Comment by theashiot** [comment]
Thanks @adahms, LGTM!

best,
ashwin

**Comment by adahms** [comment]
@redhataccess/pantheon-developers - Ready for merge.

**Review by benradey — APPROVED** [review]

**Pull request merged**